homeassistant/components/toon/climate.py

```diff
@@ -1,90 +1,151 @@
-    def supported_features(self):
-        """Return the list of supported features."""
-        return SUPPORT_FLAGS
+    def unique_id(self) -> str:
+        """Return the unique ID for this thermostat."""
+        return self.toon.agreement.id
 
     @property
-    def name(self):
-        """Return the name of this thermostat."""
-        return self._name
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this thermostat."""
+        agreement = self.toon.agreement
+        return {
+            'identifiers': {
+                (DOMAIN, agreement.id)
+            },
+            'name': self._name,
+            'manufacturer': 'Eneco',
+            'model': agreement.display_hardware_version.rpartition('/')[0],
+            'sw_version': agreement.display_software_version
+                            .rpartition('/')[-1],
+        }
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement used by the platform."""
         return TEMP_CELSIUS
 
     @property
-    def current_operation(self):
+    def current_operation(self) -> str:
         """Return current operation i.e. comfort, home, away."""
-        return TOON_HA.get(self.thermos.get_data('state'))
+        return TOON_HA.get(self._state)
 
     @property
-    def operation_list(self):
+    def operation_list(self) -> List[str]:
         """Return a list of available operation modes."""
-        return self._operation_list
+        return list(HA_TOON.keys())
 
     @property
-    def current_temperature(self):
+    def current_temperature(self) -> float:
         """Return the current temperature."""
-        return self.thermos.get_data('temp')
+        return self._current_temperature
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float:
         """Return the temperature we try to reach."""
-        return self.thermos.get_data('setpoint')
+        return self._target_temperature
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the current state of the burner."""
+        return {
+            'heating_type': self._heating_type,
+
+            'next_target_temperature': self._next_target_temperature,
+            'modulation_level': self._modulation_level,
+
+            'program_state': self._program_state,
+            'program_next': self._program_next,
+            'holiday_state': self._holiday_state,
+        }
 
-    def set_temperature(self, **kwargs):
+    def set_temperature(self, **kwargs) -> None:
         """Change the setpoint of the thermostat."""
-        temp = kwargs.get(ATTR_TEMPERATURE)
-        self.thermos.set_temp(temp)
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        self.toon.thermostat = temperature
 
```

Comment:
Should there be an min_temp and  max_temp?
The toon can only handle between 6-30 in the official app.
I do not know what happens if you send an number out of the range.

Reply:
Good point there @scheric, will look into that.

Reply:
@scheric It is indeed limited to 6-30, added that limit to the climate platform. Thanks 👍